eng/build.sh

```diff
@@ -2,63 +2,514 @@
+    "$__ProjectRoot/eng/common/build.sh" --build --configuration "$__BuildType" --verbosity "$__Verbosity" $__ManagedBuildArgs $__UnprocessedBuildArgs
+    if [ $? != 0 ]; then
+        exit 1
+    fi
+fi
+
+#
+# Initialize the target distro name
+#
+
+initTargetDistroRid
+
+echo "RID: $__DistroRid"
+
+#
+# Setup LLDB paths for native build
+#
+
+if [ "$__HostOS" == "OSX" ]; then
+    export LLDB_H=$__ProjectRoot/src/SOS/lldbplugin/swift-4.0
+    export LLDB_LIB=/Applications/Xcode.app/Contents/SharedFrameworks/LLDB.framework/LLDB
+    export LLDB_PATH=/Applications/Xcode.app/Contents/Developer/usr/bin/lldb
+
+    export MACOSX_DEPLOYMENT_TARGET=10.12
+
+    # If Xcode 9.2 exists (like on the CI/build machines), use that. Xcode 9.3 or 
+    # greater (swift 4.1 lldb) doesn't work that well (seg faults on exit).
+    if [ -f "/Applications/Xcode_9.2.app/Contents/Developer/usr/bin/lldb" ]; then
+        if [ -f "/Applications/Xcode_9.2.app/Contents/SharedFrameworks/LLDB.framework/LLDB" ]; then
+            export LLDB_PATH=/Applications/Xcode_9.2.app/Contents/Developer/usr/bin/lldb
+            export LLDB_LIB=/Applications/Xcode_9.2.app/Contents/SharedFrameworks/LLDB.framework/LLDB
+        fi
+    fi
+
+    if [ ! -f $LLDB_LIB ]; then
+        echo "Cannot find the lldb library. Try installing Xcode."
+        exit 1
+    fi
+
+    # Workaround bad python version in /usr/local/bin/python2.7 on lab machines
+    export PATH=/usr/bin:$PATH
+    which python
+    python --version
+fi
+
+#
+# Build native components
+#
+
+if [ ! -e $__DotNetCli ]; then
+   echo "dotnet cli not installed $__DotNetCli"
+   exit 1
+fi
+
+if [ $__NativeBuild == true ]; then
+    if [[ $__CI == true ]]; then
+        echo "Generating Version Source File"
+        __GenerateVersionRestoreLog="$__LogDir/GenerateVersionRestore.binlog"
+        $__DotNetCli msbuild $__ProjectRoot/eng/CreateVersionFile.csproj /v:$__Verbosity /bl:$__GenerateVersionRestoreLog /t:Restore /p:Configuration="$__BuildType" /p:Platform="$__BuildArch" $__UnprocessedBuildArgs
+
+        __GenerateVersionLog="$__LogDir/GenerateVersion.binlog"
+        $__DotNetCli msbuild $__ProjectRoot/eng/CreateVersionFile.csproj /v:$__Verbosity /bl:$__GenerateVersionLog /t:GenerateVersionFiles /p:GenerateVersionSourceFile=true /p:NativeVersionSourceFile="$__IntermediatesDir/version.cpp" /p:Configuration="$__BuildType" /p:Platform="$__BuildArch" $__UnprocessedBuildArgs
+        if [ $? != 0 ]; then
+            echo "Generating Version Source File FAILED"
+            exit 1
+        fi
+    else
+        echo "Generating Empty Version Source File"
+       echo "" > "$__IntermediatesDir/version.cpp"
+    fi
+
+    build_native "$__BuildArch" "$__IntermediatesDir" "$__ExtraCmakeArgs"
+fi
+
+#
+# Copy the native SOS binaries to where these tools expect for testing
+#
+
+if [[ $__NativeBuild == true || $__Test == true ]]; then
+    __dotnet_sos=$__RootBinDir/bin/dotnet-sos/$__BuildType/netcoreapp2.1/publish/$__DistroRid
```

Comment:
Same comment about deployment. Maybe they should be an install target in cmake, that way it covers all cases?

Reply:
You can only have one cmake install target which is the source of this copy (i.e. artifacts/bin/Linux.x64.Debug) and it needs the rid for the subdirectory which this script has already built.